packages/storage/src/indexed-db/index.ts

```diff
@@ -669,4 +674,55 @@ export class IndexedDb implements IndexedDbInterface {
       txs.add({ table: 'SWAPS', value: n.toJson() as Jsonified<SwapRecord> });
     }
   }
+
+  // As more auction types are created, add them to T as a union type.
+  async upsertAuction<T extends DutchAuctionDescription>(
+    auctionId: AuctionId,
+    value: {
+      auction?: T;
+      noteCommitment?: StateCommitment;
```

Comment:
Yeah, because we save to the `AUCTIONS` table twice: once with the auction, and once with the note commitment. It's a little awkward, but making methods specific to which value is being saved (e.g., `upsertAuctionWithNoteCommitment` felt messier.